src/widget/widget.ts

```diff
@@ -82,6 +77,29 @@ export class UbirchVerificationWidget {
     </div>`;
   }
 
+  public reset(newMessenger?: Observable<UbirchMessage | null>): void {
+    this.host.innerHTML = '';
+    this.linkToConsole = true;
+    this.openConsoleInSameTarget = false;
+    this.headlineText = '';
+    this.resultText = '';
+    this.blockchainIconsAnchors = '';
+    this.stage = EStages.prod;
+    this.messenger = newMessenger || new Observable(null);
+    this.subscribeToMessenger();
+  }
+
+  private subscribeToMessenger(): void {
+    this.messenger.subscribe((message) => {
```

Comment:
better store the subscription in variable
```
const this.subscription = this.messenger.subscribe((message) => {...
```
and then `this.subscription.unsubscribe()`  in reset method